AffirmSDK/AffirmCheckoutData.m

```diff
@@ -241,24 +258,38 @@ + (AffirmCheckout *)checkoutWithItems:(NSArray <AffirmItem *>*)items
     return [self checkoutWithItems:items shipping:shipping taxAmount:taxAmount shippingAmount:shippingAmount discounts:nil metadata:nil financingProgram:nil];
 }
 
++ (AffirmCheckout *)checkoutWithItems:(NSArray <AffirmItem *>*)items
+                             shipping:(AffirmShippingDetail *)shipping
+                            taxAmount:(NSDecimalNumber *)taxAmount
+                       shippingAmount:(NSDecimalNumber *)shippingAmount
+                          totalAmount:(NSDecimalNumber *)totalAmount {
+    return [self checkoutWithItems:items shipping:shipping taxAmount:taxAmount shippingAmount:shippingAmount discounts:nil metadata:nil financingProgram:nil totalAmount:totalAmount];
+}
+
 + (AffirmCheckout *)checkoutWithItems:(NSArray <AffirmItem *>*)items
                              shipping:(AffirmShippingDetail *)shipping
                             taxAmount:(NSDecimalNumber *)taxAmount
                        shippingAmount:(NSDecimalNumber *)shippingAmount
                             discounts:(NSArray <AffirmDiscount *>*)discounts
                              metadata:(NSDictionary *)metadata {
-    return [[self alloc] initWithItems:items shipping:shipping taxAmount:taxAmount shippingAmount:shippingAmount discounts:discounts metadata:metadata financingProgram:nil];
+    return [[self alloc] initWithItems:items shipping:shipping taxAmount:taxAmount shippingAmount:shippingAmount discounts:discounts metadata:metadata financingProgram:nil totalAmount:nil];
 }
 
 - (NSDecimalNumber *)total {
+    if (self.totalAmount) {
+        return self.totalAmount;
+    }
+    
     NSDecimalNumber *total = [self.taxAmount decimalNumberByAdding:self.shippingAmount];
     for (AffirmItem *item in self.items) {
         total = [total decimalNumberByAdding:[item.unitPrice decimalNumberByMultiplyingBy:[AffirmNumberUtils integerQuantityToDecimalNumber:item.quantity]]];
     }
     for (AffirmDiscount *discount in self.discounts) {
         total = [total decimalNumberBySubtracting:discount.amount];
     }
-    return total;
+    
+    _totalAmount = [AffirmNumberUtils decimalDollarsToIntegerCents:total];
```

Comment:
I think it's more standard to use `self.totalAmount` here. Ref: https://stackoverflow.com/questions/14112715/self-variablename-vs-variablename-vs-sysnthesize-variablename#14113098

Reply:
Agreed. It's also simpler to read.